api/scpca_portal/models/dataset.py

```diff
@@ -67,6 +80,63 @@ class Meta:
     def __str__(self):
         return f"Dataset {self.id}"
 
+    @classmethod
+    def get_ccdl_dataset(cls, ccdl_name, project_id: str | None = None) -> Self:
+        dataset = cls(is_ccdl=True, ccdl_name=ccdl_name, ccdl_project_id=project_id)
+        dataset.format = dataset.ccdl_type["format"]
+        dataset.data = dataset.get_ccdl_data()
+        return dataset
+
+    def get_ccdl_data(self) -> Dict:
+        if not self.is_ccdl:
+            raise ValueError("Invalid Dataset: Dataset must be CCDL.")
+
+        projects = Project.objects.all()
+        if self.ccdl_project_id:
+            projects = projects.filter(scpca_id=self.ccdl_project_id)
+
+        data = {}
+        for project in projects:
+            samples = project.samples
```

Comment:
Use the `all()` method to query all the associated samples here.

```suggestion
            samples = project.samples.all()
```

Reply:
Good catch